src/local-development/helpers/component-id-mapping-helper.ts

```diff
@@ -98,4 +127,55 @@ export class ComponentIdMappingHelper {
 
 		return localId;
 	}
+
+	addLocalDeleted(componentType: AppComponentType | AppGeneralType, localId: string) {
+		const mapping = this.getMappingByLocalName(componentType, localId);
+		if (!mapping) {
+			return;
+		}
+		mapping.localDeleted = true;
+	}
+
+	/**
+	 * Removes a mapping entry for a specified component type based on the remote component name.
+	 *
+	 * This function finds the specified component type within the `idMapping` of `makecomappJson`
+	 * for the given origin. If an entry with a matching remote component name exists,
+	 * it is filtered out from the mappings.
+	 *
+	 * @param {AppComponentType} componentType - The type of the component to target in the mapping.
+	 * @param {string} originComponentName - The name of the remote component to remove from the mapping.
+	 */
+	removeByRemoteName(componentType: AppComponentType, originComponentName: string) {
+		const originInMakecomappJson = getOriginObject(this.makecomappJson, this.origin);
+
+		// Modify makecomapp object
+		if (originInMakecomappJson.idMapping?.[componentType]) {
+			originInMakecomappJson.idMapping[componentType] = originInMakecomappJson.idMapping[componentType].filter(mapping => mapping.remote !== originComponentName);
+		}
+		// Modify origin object that
+		if (this.origin.idMapping?.[componentType]) {
+			this.origin.idMapping[componentType] = this.origin.idMapping[componentType].filter(mapping => mapping.remote !== originComponentName);
+		}
+
+		return originInMakecomappJson;
+	}
+
+
+	filterMappingItems(filterCondition: (it: ComponentIdMappingItem) => boolean) {
+		const originInMakecomappJson = getOriginObject(this.makecomappJson, this.origin);
+
+		for (const componentType of AppComponentTypes) {
+			// Modify makecomapp object
+			if (originInMakecomappJson.idMapping?.[componentType]) {
+				originInMakecomappJson.idMapping[componentType] = originInMakecomappJson.idMapping[componentType].filter(filterCondition);
+			}
+			// Modify origin object that
+			if (this.origin.idMapping?.[componentType]) {
+				this.origin.idMapping[componentType] = this.origin.idMapping[componentType].filter(filterCondition);
+			}
+
+		}
+		return originInMakecomappJson;
```

Comment:
Returning values if not used anywhere in implementation. Consider remove the `return`.

```suggestion
```